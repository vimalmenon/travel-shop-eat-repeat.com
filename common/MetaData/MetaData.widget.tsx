import Head from 'next/head';

import { IMetaData } from './MetaData';

export const MetaData: React.FC<IMetaData> = ({ metaData }) => {
  const title = metaData?.titles.join(metaData.titleSeparator);
  const description = metaData?.description || '';
  return (
    <Head>
      <meta charSet="utf-8" />
      <title>{title}</title>
      <meta name="description" content={description} />
      <meta name="viewport" content="minimum-scale=1, initial-scale=1, width=device-width" />
    </Head>
  );
};
