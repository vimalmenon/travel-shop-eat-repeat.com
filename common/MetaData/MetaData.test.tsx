import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MetaData } from './MetaData.widget';

describe('common > <MetaData> Component', () => {
  describe('Snapshot Testing', () => {
    it('<MetaData> WITHOUT description', () => {
      const wrapper = shallow(<MetaData title="App Title" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
    it('<MetaData> WITH description', () => {
      const wrapper = shallow(<MetaData title="App Title" description="App description" />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
