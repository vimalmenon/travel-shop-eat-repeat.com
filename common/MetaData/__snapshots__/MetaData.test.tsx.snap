// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common > <MetaData> Component Snapshot Testing <MetaData> WITH description 1`] = `
<Head>
  <meta
    charSet="utf-8"
  />
  <title>
    App Title
  </title>
  <meta
    content="App description"
    name="description"
  />
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
</Head>
`;

exports[`common > <MetaData> Component Snapshot Testing <MetaData> WITHOUT description 1`] = `
<Head>
  <meta
    charSet="utf-8"
  />
  <title>
    App Title
  </title>
  <meta
    content=""
    name="description"
  />
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
</Head>
`;
